Kudu.Core/Deployment/Generator/AspNetCoreBuilder.cs

```diff
@@ -27,7 +28,14 @@ protected override string ScriptGeneratorCommandArguments
                 }
                 else
                 {
-                    return $"--aspNetCore \"{Path.GetDirectoryName(_projectPath)}\" --solutionFile {_solutionPath}";
+                    // infer path of the ".xproj" file based on projectPath,
+                    // ".xproj" represents the web project in the solution
+                    string[] xprojFile = Directory.GetFiles(Path.GetDirectoryName(_projectPath), "*.xproj");
+                    if (xprojFile.Length != 1)
+                    {
+                        throw new InvalidOperationException($"expecting EXACTLY one .xproj file for web project, found none or several under {Path.GetDirectoryName(_projectPath)}");
```

Comment:
You probably can say `found {xprojFile.Length} file(s) under` instead of `found none or several under`